docs/madr/decisions/056-meshexternalservice-and-zoneegress.md

```diff
@@ -0,0 +1,371 @@
+It does not make sense to apply inbound policies on the sidecar since there is no inbound for it.
+Egress plays a role of a "sidecar" that is inside the mesh, and we can target it's inbound.
+Configuring outbound policies on the sidecar avoids the problem with different clients influencing each other (see [MeshCircuitBreaker](#meshcircuitbreaker)).
+
+##### How to configure it
+
+###### `to`
+
+In `to` section we want to capture all the traffic going `to` a `MeshExternalService.
+
+The situation is pretty straightforward:
+
+```yaml
+kind: MeshTimeout
+spec:
+  targetRef:
+    kind: Mesh
+    proxyType: sidecar # just to be explicit that it's on the sidecar? see section 'Configurable where it makes sense on both sidecar and egress, predefined where only one makes sense'
+  to:
+  - targetRef:
+      kind: MeshExternalService
+    default:
+      idleTimeout: 20s
+```
+
+###### `from`
+
+It never makes sense to put `MeshExternalService` in `from` section because traffic never comes from `MeshExternalService`.
+
+###### Top level
+
+We want to have the option to configure inbound policies for `MeshExternalService`.
+
+One option is to use `kind: Mesh`, `proxyType` + `sectionName` just like in [MeshService](https://github.com/kumahq/kuma/blob/7e69b872404251d4e5cd766225fdd1b821cc6c5d/docs/madr/decisions/046-meshservice-policy-matching.md?plain=1#L192):
+
+```yaml
+kind: MeshAccessLog
+spec:
+  targetRef:
+    kind: Mesh
+    proxyType: egress # when it's egress it's MES name
+    sectionName: httpbin
+  from:
+    - targetRef:
+        kind: Mesh
+      default:
+        backends:
+          - type: File
+            file:
+              path: /tmp/access.log
+```
+
+Another option is to use `MeshExternalService` as a top level target + `proxyType: egress` + `name`.
+
+```yaml
+kind: MeshAccessLog
+spec:
+  targetRef:
+    kind: MeshExternalService
+    proxyType: egress
+    name: httpbin
+  from:
+    - targetRef:
+        kind: Mesh
+      default:
+        backends:
+          - type: File
+            file:
+              path: /tmp/access.log
+```
+
+#### Only on the egress
+
+There are policies that make sense to be in both places and the ones like MeshTimeout that only have drawbacks on egress.
+
+#### Only on sidecar
+
+There are policies that make more sense on egress and in both places.
+
+#### Configurable where it makes sense on both sidecar and egress, predefined where only one makes sense
+
+We've not seen any significant advantages to putting `to` policies on egress.
+Almost all seem to suffer from the same problems as described in [MeshCircuitBreaker](#MeshCircuitBreaker).
+
+Although for MeshAccessLog we could do:
+
+```yaml
+kind: MeshAccessLog
+spec:
+  targetRef:
+    kind: Mesh
+    proxyType: egress # or `sidecar`
+  to:
+    - targetRef:
+        kind: MeshExternalService
+        name: httpbin
+      default:
+        backends:
+          - type: File
+            file:
+              path: /tmp/access.log
+```
```

Comment:
I think this is not acknowledging that there are multiple problems and we've decided not to address some of it:

Regular app:
```mermaid
flowchart LR
  SrcApp --A--> SrcSidecar --B--> Network --C--> DestSidecar --D--> DestApp 
```
MES with Egress:
```mermaid
flowchart LR
  SrcApp --A--> SrcSidecar --B--> Network --C--> Egress --D--> ExternalService
```

We see there are 4 locations where Envoy plays a role.

B is done with `to`
C is done with `from`

The targetRef inside to and from only identify the type of traffic.

There is nowhere the ability to apply policies to A and D we never really talked about it because in the regular use case this is all on localhost so it's not that useful. Here you are pointing the 1 place where maybe it would be useful, I'd vote for judging out of scope.

I think that's really all there is with policy matching.

Reply:
> Here you are pointing the 1 place where maybe it would be useful, I'd vote for judging out of scope.

👍